src/main/scala/org/bblfsh/client/libuast/alloclist.c

```diff
@@ -0,0 +1,38 @@
+#ifdef __cplusplus
```

Comment:
Minor: I don't think this code is expected to be called from C++, since this is ad-hoc code for JNI.
OTOH, if you're using C++, you already have vectors and you wouldn't need this. Can this be built with a C++ compiler instead?

Reply:
I would prefer to keep it plain C for the moment to avoid dragging the C++ std into the .so (complicating the macOS build, etc). If we need more features in the future we could always move to C++.

Reply:
I'd say having a custom vector implementation is a good enough reason, but I don't know how much it complicates the build. Anyway, going back to my original question, if we're staying in plain C, what do we need to add these C++ compatibility checks? Not that it's important, just curious.

Reply:
It's pretty common to do that in plan C; if somebody compiles the file with a C++ compiler it will wrap everything inside an "extern C" so the compiler will threat the code inside as "plain old C" (C and C++ nowadays are not 100% compatible). It's not important anyway since we specify the compiler in the build, maybe I'll remove it.

Reply:
Yes, I know what is for, that's why I asked for this particular usage, since it's plain C, not expected to be used elsewhere, and we control the compiler (that's why I said that if we're expecting to use C++ we wouldn't need this custom code in the first place ;)

Reply:
After my debugging session tracking pointers to pointers I'll probably move it to C++ and std:vector sooner than later hehe. I've not done it already because JNI has a different (better) API for C++ and I would like to migrate to it too, but it's not trivial.

Reply:
PS: I also re-added the synchonization in the Scala part now that we've a working way to do it (with the `Libuast` object but outside of it). Works well under spark, BTW it would be nice to test from Spark in Travis; I need to check how to run jupiter notebooks headless and programatically.